components/style/gecko/wrapper.rs

```diff
@@ -614,6 +618,10 @@ impl<'le> TElement for GeckoElement<'le> {
     }
 
     fn get_animation_rules(&self) -> AnimationRules {
+        if !self.may_have_animations() {
+            return AnimationRules(None, None);
```

Comment:
Right, I thought avoiding the function call will be nice there. If the compiler decides to inline the calls below, it should be able to remove the checks in the other function, so there was no loss in doing this.